### ISSUE TITLE
Remove warnings from `useDeepCompareEffect`

### ISSUE THREAD

**Issue by garrettmaring** [streamich/react-use#755]
**Is your feature request related to a problem? Please describe.**

`useDeepCompareEffect` gives warnings when using it with primitive dependencies. This is frustrating for the common case where dependencies could either `undefined` or an `object` that you'd like to be deep-compared.

**Describe the solution you'd like**

Remove warnings from `useDeepCompareEffect` *or* provide a way to suppress warnings.

**Comment by xobotyi** [comment]
Any example?

**Comment by garrettmaring** [comment]
```typescript
import React from "react";
import { useDeepCompareEffect } from "react-use";

export const useObjectDidChange = <T extends any>(didChange: () => void, dep: T) => {
   useDeepCompareEffect(() => {
    didChange();
   }, [dep]);
}

function App = () => {
  const [obj, setObj] = useState<object | undefined>(undefined);
  useObjectDidChange(() => console.log("obj changed"), obj);
  
  return <div />;
}
```

This code will log a warning whenever `obj` is set to `undefined`.

**Comment by garrettmaring** [comment]
I understand that the warning is there to help users. We should be using `useEffect` when the values of the dependency array are primitives. However, it's possible for the values of the array to change from primitive to non-primitive (as shown above). In that case, there should be no warning. It is correct to use `useDeepCompareEffect` if the value of the dependency is either a primitive or a non-primitive.

That warning is only valid if it is always primitive.

**Comment by garrettmaring** [comment]
If the above case doesn't satisfy you, I'd at least add the option to hide warnings from `react-use`. Warnings are definitely useful and I appreciate that they're in `react-use`. However, they clutter the console and, in this case, I simply don't agree with the warning but would still like to continue using `react-use`.

**Comment by streamich** [comment]
@garrettmaring I agree, I think the warnings for primitives in `useDeepCompare` (and shallow compare) should be removed.

**Comment by streamich** [comment]
But the other warning about no dependencies should probably stay.

**Comment by callmetushar123** [comment]
Is this issue open? Can anyone assign this to me?

**Comment by shrutipandya3** [comment]
Can I work on this if still open?
